tools/traffic/table/blob_metadata.go

```diff
@@ -0,0 +1,69 @@
+package table
+
+import "errors"
+
+// BlobMetadata encapsulates various information about a blob written by the traffic generator.
+type BlobMetadata struct {
+	// key of the blob, set when the blob is initially uploaded.
+	key []byte
+
+	// checksum of the blob.
+	checksum *[16]byte
+
+	// size of the blob, in bytes.
+	size uint
+
+	// blobIndex of the blob.
+	blobIndex uint
+
+	// remainingReadPermits describes the maximum number of remaining reads permitted against this blob.
+	// If -1 then an unlimited number of reads are permitted.
+	remainingReadPermits int
+}
+
+// NewBlobMetadata creates a new BlobMetadata instance. The readPermits parameter describes the maximum number of
+// remaining reads permitted against this blob. If -1 then an unlimited number of reads are permitted.
+func NewBlobMetadata(
+	key []byte,
+	checksum *[16]byte,
+	size uint,
+	blobIndex uint,
+	readPermits int) (*BlobMetadata, error) {
+
+	if readPermits == 0 {
+		return nil, errors.New("read permits must not be zero")
+	}
+
+	return &BlobMetadata{
+		key:                  key,
+		checksum:             checksum,
+		size:                 size,
+		blobIndex:            blobIndex,
+		remainingReadPermits: readPermits,
+	}, nil
+}
+
+// Key returns the key of the blob.
```

Comment:
What about making the member variables public? These getters are quite simple so seem not worth the verbosity.

Reply:
Intention was to make it possible to read these values without the capability of updating them. But maybe that's more of a java design pattern. I've made the member variables public as you suggest.

Reply:
Yea it feels quite Java-ish